ffmpeg/vf_raisr.c

```diff
@@ -263,10 +272,15 @@ static int filter_frame(AVFilterLink *inlink, AVFrame *in)
         vdt_in[p].height = plane->height;
         vdt_in[p].step = in->linesize[p];
 
+        // Get horziontal and vertical power of 2 factors
+        int vsub = p ? desc->log2_chroma_h : 0;
+        int hsub = p ? desc->log2_chroma_w : 0;
+
         // fill in the output video data type structure
         vdt_out[p].pData = out->data[p];
-        vdt_out[p].width = plane->width * raisr->ratio;
-        vdt_out[p].height = plane->height * raisr->ratio;
+        // Determine the width and height of this plane/channel
+        vdt_out[p].width = AV_CEIL_RSHIFT(out->width, hsub);
+        vdt_out[p].height = AV_CEIL_RSHIFT(out->height, vsub);
```

Comment:
Is it better to assign vdt_out[p].width/height with outlink->w/h?

Reply:
The outlink->w/h is only mean w/h of Y channel, this line mainly calculate the width/height of UV channel.